sentry-core/src/main/java/io/sentry/core/EnvelopeSender.java

```diff
@@ -99,6 +120,63 @@ private void processEnvelope(SentryEnvelope envelope) throws IOException {
         logger.log(
             SentryLevel.WARNING, "Item %d of type: %s ignored.", items, item.getHeader().getType());
       }
+
+      if (!hint.succeeded) {
+        // Failed to send an item of the envelope: Stop attempting to send the rest (an attachment without the event that created it isn't useful)
+        logger.log(SentryLevel.WARNING, "Envelope for event Id: %s had a failed capture at item %d. No more items will be sent.", envelope.getHeader().getEventId(), items);
+        break;
+      }
+      hint.resetSubmissionResult();
+    }
+  }
+
+  private static final class CachedEnvelopeHint implements Cached, Retryable, SubmissionResult, Flushable {
+    boolean retry = false;
+    boolean succeeded = false;
+
+    private CountDownLatch latch;
+    private final long timeoutMills;
+    private final @Nullable ILogger logger;
+
+    CachedEnvelopeHint(final long timeoutMills, final @Nullable ILogger logger) {
+      this.timeoutMills = timeoutMills;
+      this.latch = new CountDownLatch(1);
+      this.logger = logger;
+    }
+
+    void waitFlush() {
+      try {
+        latch.await(timeoutMills, TimeUnit.MILLISECONDS);
+      } catch (InterruptedException e) {
+        logIfNotNull(
+          logger, ERROR, "Exception while awaiting for flush in UncaughtExceptionHint", e);
+      }
+    }
+
+    public void newCountDown() {
+      latch = new CountDownLatch(1);
+    }
+
+    @Override
+    public void flushed() {
```

Comment:
is it `flushed` a good name? what's about only `flush`?

Reply:
`flush` seems like the action of flushing.
I need `markFlushed` maybe a better name then